### PR TITLE
Solo editing perf track local branch array

### PR DESCRIPTION
**Pull request by yann-achard-MS**
## Description

Before this PR, the "EditManager - Bench No concurrency Many local edits" benchmark reports the following times on a dev workstation:
```
status  name                                          period (ns/op)  relative margin of error  iterations per batch  batch count  total time (s)
------  --------------------------------------------  --------------  ------------------------  --------------------  -----------  --------------
    ✔   Process the sequencing of 1 local commits           1,317.27                    ±1.42%                     1       21,286            5.00
    ✔   Process the sequencing of 10 local commits          9,143.64                    ±1.00%                     1       11,520            0.92
    ✔   Process the sequencing of 100 local commits       356,412.61                    ±1.00%                     1          912            0.45
    ✔   Process the sequencing of 1000 local commits   27,710,768.29                    ±0.98%                     1           41            1.24
```

After this PR:
```
status  name                                          period (ns/op)  relative margin of error  iterations per batch  batch count  total time (s)
------  --------------------------------------------  --------------  ------------------------  --------------------  -----------  --------------
    ✔   Process the sequencing of 1 local commits           1,163.39                    ±2.21%                     1       19,962            5.00
    ✔   Process the sequencing of 10 local commits          4,308.33                    ±1.94%                     1       20,516            5.00
    ✔   Process the sequencing of 100 local commits        44,794.80                    ±1.00%                     1        5,599            0.95
    ✔   Process the sequencing of 1000 local commits      574,640.66                    ±1.00%                     1        2,319            3.26
```

## Changes

The `EditManager` now maintains an array of commits that represents the changes on the local branch that are not yet on the trunk. This saves the `EditManager` manager from having to re-crawl the local branch every time a commit from the local branch is sequenced.

# Breaking Change
No breaking changes

### CODE REVIEW

File: packages/dds/tree/src/test/shared-tree-core/edit-manager/editManagerCorrectness.test.ts [hunk 1]
```diff
@@ -321,7 +321,7 @@ export function testCorrectness() {
 					//         └─ <- fork1
 
 					manager.advanceMinimumSequenceNumber(brand(2));
-					// Advancing the minimum sequence number does not evict any commits because fork1 branches off of the trunk before commit 1.
+					// Advancing the minimum sequence number does not evict any commits because fork1 branches off of the trunk at commit 1.
```
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree-core/edit-manager/editManagerCorrectness.test.ts:324]: This change is unrelated. I forgot to commit it in the previous PR.

File: packages/dds/tree/src/shared-tree-core/editManager.ts [hunk 2]
```diff
@@ -531,7 +550,7 @@ export class EditManager<
 	}
 
 	public getLocalCommits(): readonly RecursiveReadonly<GraphCommit<TChangeset>>[] {
-		return getPathFromBase(this.localBranch.getHead(), this.trunk.getHead());
+		return this.localCommits;
```
Comment (alex-pardes) [inline packages/dds/tree/src/shared-tree-core/editManager.ts:553]: Is this the reverse of the previous order?

Comment (alex-pardes) [reply packages/dds/tree/src/shared-tree-core/editManager.ts:553]: Nevermind. I misread the comment on `localCommits`

**Review by alex-pardes — APPROVED** [review]
